write-fonts/src/tables/variations.rs

```diff
@@ -21,6 +21,237 @@ impl VariationRegionList {
     }
 }
 
+/// <https://learn.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats#packed-point-numbers>
+pub struct PackedPointNumbers {
+    is_all: bool,
+    numbers: Vec<u16>,
+}
+
+/// <https://learn.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats#packed-deltas>
+pub struct PackedDeltas {
+    deltas: Vec<i16>,
+}
+
+impl Validate for PackedDeltas {
+    fn validate_impl(&self, _ctx: &mut ValidationCtx) {}
+}
+
+impl FontWrite for PackedDeltas {
+    fn write_into(&self, writer: &mut TableWriter) {
+        for run in self.iter_runs() {
+            run.write_into(writer)
+        }
+    }
+}
+
+impl PackedDeltas {
+    /// Construct a `PackedDeltas` from a vector of raw delta values.
+    pub fn new(deltas: Vec<i16>) -> Self {
+        Self { deltas }
+    }
+
+    fn iter_runs(&self) -> impl Iterator<Item = PackedDeltaRun> {
+        const MAX_POINTS_PER_RUN: usize = 64;
+
+        fn in_i8_range(val: i16) -> bool {
+            const MIN: i16 = i8::MIN as i16;
+            const MAX: i16 = i8::MAX as i16;
+            (MIN..=MAX).contains(&val)
+        }
```

Comment:
this might be just me but ... I really dislike nesting pure functions. It makes it much harder to read the code.

Reply:
funny I actually prefer this style when it isn't too long because I find it easier to read, since the functions are declared right where they're used, instead of being mysterious helpers drifting to the bottom of the file.

fwiw this pattern is used quite a bit in std:
- simple example https://doc.rust-lang.org/1.68.2/src/alloc/vec/mod.rs.html#1406
- more extreme example, declaring a helper struct & impl block inside a fn: https://doc.rust-lang.org/1.68.2/src/alloc/vec/mod.rs.html#1609

I guess how I'd think about this is that it makes the particular function a bit harder to read, at the cost of making the file overall easier to read.

Reply:
Also we're kind of reinventing TryFrom so unless there is a perf issue I wonder if we should just use that?

Reply:
I don't think that `TryFrom` should be used *anytime* you have a fallible constructor that takes a single argument. [Per the docs](http://doc.rust-lang.org/1.68.2/std/convert/trait.TryFrom.html), `TryFrom` is for "Simple and safe type conversions that may fail in a controlled way under certain circumstances". If you look at the impls in `std` for `TryFrom`, you will see that they are all very simple conversions, with clear failure conditions.